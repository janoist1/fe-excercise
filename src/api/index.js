const API_URL = 'http://localhost:3001/data';

export const fetchServiceUpdates = () =>
  fetch(API_URL)
    .then(
      response =>
        new Promise(resolve => setTimeout(() => resolve(response), 1000))
    )
    .then(response => response.json());
