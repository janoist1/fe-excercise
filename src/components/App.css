.App {
  font-family: Helvetica;
  margin: 15px 5px;
}

.App__header {
  margin-left: 55px;
  position: relative;
}

.App__header::before {
  position: absolute;
  left: -55px;
  font-size: 48px;
  content: 'ðŸš†';
}

.App__from,
.App__to {
  margin: 5px 0;
}

.App__operated-by {
  font-size: 14px;
}

@media (min-width: 425px) {
  .App {
    font-family: Helvetica;
    margin: 25px 10px;
  }

  .App__header {
    margin-left: 75px;
    margin-bottom: 35px;
    font-size: 18px;
  }

  .App__header::before {
    position: absolute;
    left: -75px;
    font-size: 72px;
    content: 'ðŸš†';
  }


  .App__operated-by {
    font-size: 16px;
  }
}
