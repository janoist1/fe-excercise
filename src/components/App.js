import React, { Component } from 'react';
import { fetchServiceUpdates } from '../api';
import ServiceUpdates from './ServiceUpdates';
import './App.css';

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      data: null
    };
  }

  componentDidMount() {
    this.loadData();
  }

  loadData() {
    return fetchServiceUpdates().then(data => {
      this.setState({ data });
    });
  }

  renderLoading() {
    return <div>Loading ...</div>;
  }

  render() {
    const { data } = this.state;

    if (!data) {
      return this.renderLoading();
    }

    const { callingPoints } = data;

    return (
      <div className="App">
        <header className="App__header">
          <h3 className="App__from">
            {callingPoints[0].station}
          </h3>
          <h3 className="App__to">
            <span>to</span> {callingPoints.slice(-1)[0].station}
          </h3>
          <span className="App__operated-by">Operated by Thameslink</span>
        </header>
        <main>
          <ServiceUpdates>
            {callingPoints}
          </ServiceUpdates>
        </main>
        <footer>by janoist1@gmail.com</footer>
      </div>
    );
  }
}

export default App;
