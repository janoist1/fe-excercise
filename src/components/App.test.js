import { shallow } from 'enzyme';
import React from 'react';
import * as api from '../api';
import App from './App';
import { data as mockData } from '../../data/ldb.json';


describe('<App />', () => {
  it('renders loading', () => {
    const component = shallow(<App />);

    expect(component.html()).toMatchSnapshot();
  });

  it('renders loaded app', () => {
    const component = shallow(<App />);

    component.instance().setState({ data: mockData });

    expect(component.html()).toMatchSnapshot();
  });

  it('calls api on mount', done => {
    api['fetchServiceUpdates'] = jest.fn(() => Promise.resolve(mockData));

    const component = shallow(<App />);

    component.instance().componentDidMount();

    process.nextTick(() => {
      expect(api.fetchServiceUpdates.mock.calls.length).toBe(1);

      done();
    });
  });
});
