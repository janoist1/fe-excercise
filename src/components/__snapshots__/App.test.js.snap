exports[`<App /> renders loaded app 1`] = `"<div class=\"App\"><header class=\"App__header\"><h3 class=\"App__from\">Sevenoaks</h3><h3 class=\"App__to\"><span>to</span> West Hampstead Thameslink</h3><span class=\"App__operated-by\">Operated by Thameslink</span></header><main><ul class=\"ServiceUpdates\"><li><div class=\"CallingPoint -departed\"><time class=\"CallingPoint__scheduled-at\">10:00</time><div class=\"CallingPoint__indicator -departed -origin\"></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">Sevenoaks</dt><dd class=\"CallingPoint__estimated-at\">Dept. On time</dd></dl></div></li><li><div class=\"CallingPoint -departed\"><time class=\"CallingPoint__scheduled-at\">10:03</time><div class=\"CallingPoint__indicator -departed\"></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">Bat &amp; Ball</dt><dd class=\"CallingPoint__estimated-at\">Dept. On time</dd></dl></div></li><li><div class=\"CallingPoint -departed\"><time class=\"CallingPoint__scheduled-at\">10:07</time><div class=\"CallingPoint__indicator -departed\"></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">Otford</dt><dd class=\"CallingPoint__estimated-at\">Dept. 10:10</dd></dl></div></li><li><div class=\"CallingPoint -departed\"><time class=\"CallingPoint__scheduled-at\">10:10</time><div class=\"CallingPoint__indicator -departed\"></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">Shoreham (Kent)</dt><dd class=\"CallingPoint__estimated-at\">Dept. 10:14</dd></dl></div></li><li><div class=\"CallingPoint -departed\"><time class=\"CallingPoint__scheduled-at\">10:13</time><div class=\"CallingPoint__indicator -departed\"></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">Eynsford</dt><dd class=\"CallingPoint__estimated-at\">Dept. 10:18</dd></dl></div></li><li><div class=\"CallingPoint -departed\"><time class=\"CallingPoint__scheduled-at\">10:18</time><div class=\"CallingPoint__indicator -departed\"></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">Swanley</dt><dd class=\"CallingPoint__estimated-at\">Dept. 10:23</dd></dl></div></li><li><div class=\"CallingPoint -departed\"><time class=\"CallingPoint__scheduled-at\">10:22</time><div class=\"CallingPoint__indicator -departed\"></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">St Mary Cray</dt><dd class=\"CallingPoint__estimated-at\">Dept. 10:27</dd></dl></div></li><li><div class=\"CallingPoint -departed\"><time class=\"CallingPoint__scheduled-at\">10:27</time><div class=\"CallingPoint__indicator -departed\"></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">Bickley</dt><dd class=\"CallingPoint__estimated-at\">Dept. 10:37</dd></dl></div></li><li><div class=\"CallingPoint -departed\"><time class=\"CallingPoint__scheduled-at\">10:30</time><div class=\"CallingPoint__indicator -departed\"></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">Bromley South</dt><dd class=\"CallingPoint__estimated-at\">Dept. 10:39</dd></dl></div></li><li><div class=\"CallingPoint -departed\"><time class=\"CallingPoint__scheduled-at\">10:32</time><div class=\"CallingPoint__indicator -departed\"></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">Shortlands</dt><dd class=\"CallingPoint__estimated-at\">Dept. 10:42</dd></dl></div></li><li><div class=\"CallingPoint -departed\"><time class=\"CallingPoint__scheduled-at\">10:35</time><div class=\"CallingPoint__indicator -departed\"></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">Ravensbourne</dt><dd class=\"CallingPoint__estimated-at\">Dept. 10:45</dd></dl></div></li><li><div class=\"CallingPoint -departed\"><time class=\"CallingPoint__scheduled-at\">10:37</time><div class=\"CallingPoint__indicator -departed\"></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">Beckenham Hill</dt><dd class=\"CallingPoint__estimated-at\">Dept. 10:47</dd></dl></div></li><li><div class=\"CallingPoint -departed\"><time class=\"CallingPoint__scheduled-at\">10:39</time><div class=\"CallingPoint__indicator -departed\"></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">Bellingham</dt><dd class=\"CallingPoint__estimated-at\">Dept. 10:49</dd></dl></div></li><li><div class=\"CallingPoint -departed\"><time class=\"CallingPoint__scheduled-at\">10:42</time><div class=\"CallingPoint__indicator -departed\"></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">Catford</dt><dd class=\"CallingPoint__estimated-at\">Dept. 10:52</dd></dl></div></li><li><div class=\"CallingPoint -departed\"><time class=\"CallingPoint__scheduled-at\">10:44</time><div class=\"CallingPoint__indicator -departed -train-here\"><div class=\"CallingPoint__train -departed\"></div></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">Crofton Park</dt><dd class=\"CallingPoint__estimated-at\">Dept. 10:55</dd></dl></div></li><li><div class=\"CallingPoint\"><time class=\"CallingPoint__scheduled-at\">10:47</time><div class=\"CallingPoint__indicator\"></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">Nunhead</dt><dd class=\"CallingPoint__estimated-at\">Exp. 10:57</dd></dl></div></li><li><div class=\"CallingPoint\"><time class=\"CallingPoint__scheduled-at\">10:50</time><div class=\"CallingPoint__indicator\"></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">Peckham Rye</dt><dd class=\"CallingPoint__estimated-at\">Exp. 11:00</dd></dl></div></li><li><div class=\"CallingPoint\"><time class=\"CallingPoint__scheduled-at\">10:54</time><div class=\"CallingPoint__indicator\"></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">Denmark Hill</dt><dd class=\"CallingPoint__estimated-at\">Exp. 11:03</dd></dl></div></li><li><div class=\"CallingPoint\"><time class=\"CallingPoint__scheduled-at\">11:00</time><div class=\"CallingPoint__indicator\"></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">Elephant &amp; Castle</dt><dd class=\"CallingPoint__estimated-at\">Exp. 11:09</dd></dl></div></li><li><div class=\"CallingPoint\"><time class=\"CallingPoint__scheduled-at\">11:04</time><div class=\"CallingPoint__indicator\"></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">London Blackfriars</dt><dd class=\"CallingPoint__estimated-at\">Exp. 11:13</dd></dl></div></li><li><div class=\"CallingPoint\"><time class=\"CallingPoint__scheduled-at\">11:06</time><div class=\"CallingPoint__indicator\"></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">City Thameslink</dt><dd class=\"CallingPoint__estimated-at\">Exp. 11:15</dd></dl></div></li><li><div class=\"CallingPoint\"><time class=\"CallingPoint__scheduled-at\">11:10</time><div class=\"CallingPoint__indicator\"></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">Farringdon Underground</dt><dd class=\"CallingPoint__estimated-at\">Exp. 11:18</dd></dl></div></li><li><div class=\"CallingPoint\"><time class=\"CallingPoint__scheduled-at\">11:14</time><div class=\"CallingPoint__indicator\"></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">London St Pancras (Intl)</dt><dd class=\"CallingPoint__estimated-at\">Exp. 11:21</dd></dl></div></li><li><div class=\"CallingPoint\"><time class=\"CallingPoint__scheduled-at\">11:18</time><div class=\"CallingPoint__indicator\"></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">Kentish Town</dt><dd class=\"CallingPoint__estimated-at\">Exp. 11:25</dd></dl></div></li><li><div class=\"CallingPoint\"><time class=\"CallingPoint__scheduled-at\">11:23</time><div class=\"CallingPoint__indicator -destination\"></div><dl class=\"CallingPoint__status\"><dt class=\"CallingPoint__station\">West Hampstead Thameslink</dt><dd class=\"CallingPoint__estimated-at\">Exp. 11:29</dd></dl></div></li></ul></main><footer>by janoist1@gmail.com</footer></div>"`;

exports[`<App /> renders loading 1`] = `"<div>Loading ...</div>"`;
